File: lib/domain/deletion_undo/exceptions.rb
# frozen_string_literal: true

module Domain
  module DeletionUndo
    module Exceptions
      class DeletionUndoError < StandardError; end
      class DeletionUndoNotFoundError < DeletionUndoError; end
      class DeletionUndoExpiredError < DeletionUndoError; end
      class DeletionUndoRestoreConflictError < DeletionUndoError; end
    end
  end
end

File: crates/agrr-domain/src/deletion_undo/exceptions.rs
//! Ruby: `Domain::DeletionUndo::Exceptions`

use thiserror::Error;

/// Ruby: `DeletionUndoError`
#[derive(Debug, Error, PartialEq, Eq)]
#[error("{0}")]
pub struct DeletionUndoError(pub String);

/// Ruby: `DeletionUndoNotFoundError`
#[derive(Debug, Error, PartialEq, Eq)]
#[error("deletion undo not found")]
pub struct DeletionUndoNotFoundError;

/// Ruby: `DeletionUndoExpiredError`
#[derive(Debug, Error, PartialEq, Eq)]
#[error("{0}")]
pub struct DeletionUndoExpiredError(pub String);

/// Ruby: `DeletionUndoRestoreConflictError`
#[derive(Debug, Error, PartialEq, Eq)]
#[error("{0}")]
pub struct DeletionUndoRestoreConflictError(pub String);
